torch/csrc/autograd/function_hook.h

```diff
@@ -2,26 +2,38 @@
 
 #include <ATen/Tensor.h>
 #include <torch/csrc/Export.h>
+#include <string>
 #include <vector>
 
 // A hook that's called on gradients
 
 namespace torch {
 namespace autograd {
 
+class CompiledNodeArgs;
 using Variable = at::Tensor;
 using variable_list = std::vector<Variable>;
 
 struct TORCH_API FunctionPreHook {
   virtual ~FunctionPreHook() = default;
   virtual variable_list operator()(const variable_list& grads) = 0;
+  virtual void compiled_args(CompiledNodeArgs& args) {
```

Comment:
I guess these are the same as the function.h version? Can we have more detailed doc on what this is supposed to do and how?

Reply:
yes, will update